pkg/apply/prune/prune.go

```diff
@@ -99,6 +99,16 @@ func (p *Pruner) Prune(
 	for _, obj := range objs {
 		id := object.UnstructuredToObjMetadata(obj)
 		klog.V(5).Infof("evaluating prune filters (object: %q)", id)
+
+		// UID will change if the object is deleted and re-created.
+		uid := obj.GetUID()
+		if uid == "" {
```

Comment:
I'm pretty sure we always GET the object before DELETE. So the UID should always be populated, right?